pkg/kernel/networkservice/ethernetcontext/vf_common.go

```diff
@@ -112,13 +113,59 @@ func vfCreate(vfConfig *vfconfig.VFConfig, conn *networkservice.Connection, isCl
 			if err = netlink.LinkSetVfHardwareAddr(pfLink, vfConfig.VFNum, macAddr); err != nil {
 				return errors.Wrapf(err, "failed to set MAC address for the VF: %v", macAddr)
 			}
+			log.FromContext(ctx).
+				WithField("pfLink", pfLink.Attrs().Name).
+				WithField("vf", vfConfig.VFNum).
+				WithField("MACAddr", macAddrString).
+				WithField("duration", time.Since(now)).
+				WithField("netlink", "LinkSetVfHardwareAddr").Debug("completed")
 		}
 		if vlanTag := int(ethernetContext.GetVlanTag()); vlanTag != 0 {
+			now := time.Now()
 			if err = netlink.LinkSetVfVlan(pfLink, vfConfig.VFNum, vlanTag); err != nil {
 				return errors.Wrapf(err, "failed to set VLAN for the VF: %v", vlanTag)
 			}
+			log.FromContext(ctx).
+				WithField("pfLink", pfLink.Attrs().Name).
+				WithField("vf", vfConfig.VFNum).
+				WithField("vlan", vlanTag).
+				WithField("duration", time.Since(now)).
+				WithField("netlink", "LinkSetVfVlan").Debug("completed")
 		}
 	}
 
 	return nil
 }
+
+var defaultMAC = make([]byte, 32)
+
+func vfCleanup(ctx context.Context, vfConfig *vfconfig.VFConfig) error {
+	pfLink, err := netlink.LinkByName(vfConfig.PFInterfaceName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get PF network interface: %v", vfConfig.PFInterfaceName)
+	}
+
+	now := time.Now()
+	if err = netlink.LinkSetVfHardwareAddr(pfLink, vfConfig.VFNum, defaultMAC); err != nil {
+		return errors.Wrap(err, "failed to set zero MAC address for the VF")
+	}
+	log.FromContext(ctx).
+		WithField("pfLink", pfLink.Attrs().Name).
+		WithField("vf", vfConfig.VFNum).
+		WithField("MACAddr", "00:00:00:00:00:00").
```

Comment:
I see that we're uzing zero mac, but the field is static and in some moment it could be changed. So I strongly recomend do one of two options:

1. Use defaultMAC value in funciton WIthField.
2. Pass make([]byte, 32) inline into netlink.LinkSetVfHardwareAddr(...)

Reply:
Thanks, fixed